test/metabase/util/debugging_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns metabase.util.debugging-test
+  (:require  [clojure.test :refer :all]
+             [metabase.config :as config]))
+
+(def dbg-prn-used? (atom false))
+
+(defn dbg-prn*
+  "Implementation for the `dbg-prn` macro."
+  [p x]
+  (p x)
+  x)
```

Comment:
take a look at https://github.com/metabase/metabase/blob/master/src/metabase/util.clj#L181-L210 - it's basically built to do something like this. You can reuse it.

Reply:
I'm aware of it, but I didn't see the point in adding an extra token and inverting the order; it'd be:

```clojure
(prog1 x
  (p x))
  ``

Reply:
fair point, this is a lot like `log/spy` works, which can be frustrating because of arg order. like, you can't use `log/spy` in a `->` threading macro

Reply:
at my last job we had a macro log/spy that had the right arg orders. and just a precommit hook to make sure you weren't committing it.

Reply:
I put `p` before `x` in the arglist so that

```
(->> foo
  (prn-dbg println)
  bar)
```

and

```
(-> foo
  ((partial prn-dbg println))
  bar)
```
  
  can both be used